dotnet/tests/MicroFocus.CafApi.QueryBuilder.Vqll.Builder.Internal.Tests/FilterWriterTest.cs

```diff
@@ -0,0 +1,3719 @@
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteEndArray());
+
+            _jsonBuilder.VerifyNoOtherCalls();
+
+            FilterWriter.WriteToJsonArray(clause, _jsonBuilder.Object);
+        }
+
+        [Fact]
+        public void TestTransformValueBetweenNumbersAsString()
+        {
+            Filter<string> clause = FilterFactory.Between("DATABATCH_ID", "1", "5");
+
+            var sequence = new MockSequence();
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteStartArray());
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteString("between"));
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteString("DATABATCH_ID"));
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteString("1"));
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteString("5"));
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteEndArray());
+
+            _jsonBuilder.VerifyNoOtherCalls();
+
+            FilterWriter.WriteToJsonArray(clause, _jsonBuilder.Object);
+        }
+
+        [Fact]
+        public void TestTransformValueBetweenStringAndNull()
+        {
+            Filter<string> clause = FilterFactory.Between("DATABATCH_ID", "test", null);
+
+            var sequence = new MockSequence();
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteStartArray());
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteString("between"));
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteString("DATABATCH_ID"));
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteString("test"));
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteNull());
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteEndArray());
+
+            _jsonBuilder.VerifyNoOtherCalls();
+
+            FilterWriter.WriteToJsonArray(clause, _jsonBuilder.Object);
+        }
+
+        [Fact]
+        public void TestTransformValueBetweenNullAndString()
+        {
+            Filter<string> clause = FilterFactory.Between("DATABATCH_ID", null, "test");
+
+            var sequence = new MockSequence();
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteStartArray());
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteString("between"));
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteString("DATABATCH_ID"));
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteNull());
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteString("test"));
+            _jsonBuilder.InSequence(sequence).Setup(x => x.WriteEndArray());
+
+            _jsonBuilder.VerifyNoOtherCalls();
+
+            FilterWriter.WriteToJsonArray(clause, _jsonBuilder.Object);
+        }
+
+        [Fact]
+        public void TestTransformValueBetweenStringNulls()
+        {
+            string startValue = null;
+            string endValue = null;
```

Comment:
In .net standard 2.0, which is what the libraries are written in for maximum compatibility, `string` is nullable, but in .net 6.0, you should really use `string?` for nullable strings.  (You are using a newer version of C# to run the tests.)

```suggestion
            string? startValue = null;
            string? endValue = null;
```